### PR TITLE
Values are not respected in variables

### PR DESCRIPTION
**Pull request by robinhoodie0823**
https://github.com/tokens-studio/figma-plugin/assets/103296157/29107673-ab3d-4775-81c3-129d575eb38b

### CODE REVIEW

File: src/app/components/MoreButton/MoreButton.tsx [hunk 1]
```diff
@@ -69,6 +69,7 @@ export const MoreButton: React.FC<Props> = ({ token, type, showForm }) => {
   const visibleProperties = React.useMemo(() => properties.filter((p) => p.label), [properties]);
 
   const handleEditClick = React.useCallback(() => {
+    console.log('handleEditClick in MoreButton');
```
Comment (SorsOps) [inline src/app/components/MoreButton/MoreButton.tsx:72]: Remove

Comment (six7) [inline src/app/components/MoreButton/MoreButton.tsx:72]: log

File: src/app/store/useTokens.tsx [hunk 2]
```diff
@@ -399,6 +399,16 @@ export default function useTokens() {
       name: BackgroundJobs.UI_CREATEVARIABLES,
       isInfinite: true,
     });
+
+    Object.values(tokens).forEach((tokenItem) => {
```
Comment (SorsOps) [inline src/app/store/useTokens.tsx:403]: This is mutating the data which we shouldn't be doing as then the tokens are changed when using the createVariables side effect. If we're going to make changes here for the create local variables usage we should be cloning the tokens to not get any unintentional changes in the data elsewhere

File: src/plugin/figmaTransforms/colors.ts [hunk 3]
```diff
@@ -91,6 +91,7 @@ export function convertToFigmaColor(input: string) {
   let color: RGBA;
   let opacity;
   if (input.startsWith('rgb')) {
+    console.log('starts with rgb');
```
Comment (SorsOps) [inline src/plugin/figmaTransforms/colors.ts:94]: Remove console logs

File: src/utils/TokenResolver.ts [hunk 4]
```diff
@@ -118,7 +118,6 @@ class TokenResolver {
 
         // We need to avoid circular references, so we check if we already resolved this reference
         if (resolvedReferences.has(path)) {
-          console.log('Circular reference detected:', path);
```
Comment (SorsOps) [inline src/utils/TokenResolver.ts:121]: Restore this to help with debugging

File: src/app/store/useTokens.tsx [hunk 5]
```diff
@@ -399,6 +399,16 @@ export default function useTokens() {
...
+      for (let i = 0; i < tokenItem.length; i += 1) {
```
Comment (six7) [inline src/app/store/useTokens.tsx:404]: I don't think this is how it should be done. https://github.com/tokens-studio/figma-plugin/pull/2366/files#diff-874132b880f3e22690900909f69048afec5f0ec0137e7ae8dbe6c56cbb2628f3R301-R307

here's where we fetch the resolved tokens from the store, no need to manually iterate over them

File: src/plugin/figmaTransforms/colors.ts (hunk 3 above)
Comment (six7) [inline src/plugin/figmaTransforms/colors.ts:94]: log

File: src/plugin/figmaTransforms/colors.ts [hunk 6]
```diff
@@ -101,6 +102,7 @@ export function convertToFigmaColor(input: string) {
     };
     opacity = Number(a);
   } else if (input.startsWith('hsl')) {
+    console.log('starts with hsl');
```
Comment (six7) [inline src/plugin/figmaTransforms/colors.ts:105]: log

File: src/plugin/figmaTransforms/colors.ts [hunk 7]
```diff
@@ -111,6 +113,7 @@ export function convertToFigmaColor(input: string) {
     };
     opacity = Number(a);
   } else {
+    console.log('else case');
```
Comment (six7) [inline src/plugin/figmaTransforms/colors.ts:116]: log

**Review by six7 — CHANGES_REQUESTED** [review]
can you also link the issue?

File: src/app/store/useTokens.tsx [hunk 8]
```diff
@@ -399,6 +399,16 @@ export default function useTokens() {
...
+        const resolvedValue = getAliasValue(tokenItem[i], tokensContext.resolvedTokens);
+        if (typeof resolvedValue === 'string') {
+          tokenItem[i].value = resolvedValue;
+        }
+      }
+    });
+
     const createVariableResult = await wrapTransaction({
```
Comment (SorsOps) [inline src/app/store/useTokens.tsx:412]: Can we get a test on this to verify its working ?

File: src/app/store/useTokens.test.tsx [hunk 9]
```diff
@@ -581,7 +581,7 @@ describe('useToken test', () => {
           { name: 'spacing-3', value: '{base.base-50}', type: TokenTypes.SPACING },
           { name: 'borderWidth-3', value: '{base.base-187}', type: TokenTypes.BORDER_WIDTH },
           { name: 'text', value: 'text value', type: TokenTypes.TEXT },
-          { name: 'color', value: 'rgba(20, 80, 200, 1)', type: TokenTypes.COLOR },
+          { name: 'color', value: '#1450c8', type: TokenTypes.COLOR },
```
Comment (SorsOps) [inline src/app/store/useTokens.test.tsx:584]: Why did this value change?

File: src/app/store/useTokens.tsx [hunk 10]
```diff
@@ -397,8 +397,11 @@ export default function useTokens() {
   const filterMultiValueTokens = useCallback(() => {
     const tempTokens = Object.entries(tokens).reduce((tempTokens, [tokenSetKey, tokenList]) => {
       const filteredTokenList = tokenList.filter((tokenItem) => {
+        const resolvedValue = getAliasValue(tokenItem, tokensContext.resolvedTokens) || '';
```
Comment (SorsOps) [inline src/app/store/useTokens.tsx:400]: I still can't really see a new test that confirms this behaviour ?

**Review by SorsOps — APPROVED** [review]

**Review by six7 — APPROVED** [review]

**Pull request merged**